test/mitmproxy/test_taddons.py

```diff
@@ -11,6 +11,8 @@ def test_recordingmaster():
         ctx.log.error("foo")
         assert not tctx.master.has_log("foo", level="debug")
         assert tctx.master.has_log("foo", level="error")
+        assert tctx.master.has_log("Permission denied", level="error")
+        assert tctx.master.has_log("Is a directory", level="error")
```

Comment:
I think you need to actually trigger it somehow. Right now nothing causes the error messages to be logged.
Please create a new test function in the appropriate addon test file and write a few lines of code to trigger the exception.

Reply:
For the export addon you can add something here: https://github.com/mitmproxy/mitmproxy/blob/master/test/mitmproxy/addons/test_export.py#L80

Reply:
Thanks, I get it now.

Though I am not sure what's the best way to raise a `PermissionError`, should we just attempt to write the flow in `/root/flow`? What happens when someone runs the test suite as root? Should the test just fail?

Also, sorry but should we write tests for console addons in `mitmproxy/tools/console/consoleaddons.py`?